sql/moz-fx-data-shared-prod/mozilla_vpn_derived/site_metrics_summary_v1/query.sql

```diff
@@ -1,82 +1,79 @@
+    ga_sessions.trafficSource.source,
+    ga_sessions.trafficSource.medium,
+    ga_sessions.trafficSource.campaign,
+    ga_sessions.trafficSource.adcontent AS content,
+    site,
+    ga_sessions.totals.visits,
+    ARRAY(
+      SELECT AS STRUCT
+        type AS hit_type,
+        eventInfo.eventCategory AS hit_event_category,
+        eventInfo.eventAction AS hit_event_action,
+        eventInfo.eventLabel AS hit_event_label,
+        IF(
+          site = "vpn.mozilla.org",
+          STRUCT(
+            CAST(NULL AS STRING) AS source_param,
+            eventInfo.eventLabel LIKE "%Subscribe%" AS subscribe_intent,
+            eventInfo.eventLabel LIKE "%Join%" AS join_waitlist_intent,
+            eventInfo.eventLabel LIKE "%Submit waitlist form%" AS join_waitlist_success,
+            eventInfo.eventLabel LIKE "%Sign in%" AS sign_in_intent,
+            eventInfo.eventLabel LIKE "%Download%" AS download_intent,
+            eventInfo.eventLabel LIKE "%Download installer%" AS download_installer_intent
+          ),
+          STRUCT(
+            REGEXP_EXTRACT(page.pagePath, "[?&]source=((?:whatsnew|welcome)[^&]+)") AS source_param,
+            eventInfo.eventAction = "cta: fxa-vpn"
+            AND (
+              eventInfo.eventLabel = "Try Mozilla VPN"
+              OR eventInfo.eventLabel LIKE "Get Mozilla VPN %"
+            ) AS subscribe_intent,
+            eventInfo.eventAction = "cta: button"
+            AND eventInfo.eventLabel = "Join the VPN Waitlist" AS join_waitlist_intent,
+            eventInfo.eventAction = "newsletter subscription"
+            AND eventInfo.eventLabel = "guardian-vpn-waitlist" AS join_waitlist_success,
+            eventInfo.eventAction = "cta: fxa-vpn"
+            AND eventInfo.eventLabel = "VPN Sign In" AS sign_in_intent,
+                -- Downloading clients does not happen on mozilla.org
+            FALSE AS download_intent,
+            FALSE AS download_installer_intent
+          )
+        ).*
+      FROM
+        UNNEST(ga_sessions.hits)
+      WHERE
+        site = "vpn.mozilla.org"
+        OR (site = "mozilla.org" AND page.pagePath LIKE "%/products/vpn/%")
+    ) AS hits
+  )
+);
+
+WITH ga_sessions AS (
   SELECT
-    ga_sessions.* REPLACE (SAFE.PARSE_DATE("%Y%m%d", _TABLE_SUFFIX) AS `date`),
-    "mozilla.org" AS site,
+    -- keep date out of UDF to preserve suffix pruning
+    SAFE.PARSE_DATE("%Y%m%d", _TABLE_SUFFIX) AS `date`,
+    udf_ga_sessions(ga_sessions, "mozilla.org").*,
   FROM
     `moz-fx-data-marketing-prod.65789850.ga_sessions_*` AS ga_sessions
   UNION ALL
   SELECT
-    ga_sessions.* REPLACE (SAFE.PARSE_DATE("%Y%m%d", _TABLE_SUFFIX) AS `date`),
-    "vpn.mozilla.org" AS site,
+    -- keep date out of UDF to preserve suffix pruning
+    SAFE.PARSE_DATE("%Y%m%d", _TABLE_SUFFIX) AS `date`,
+    udf_ga_sessions(ga_sessions, "vpn.mozilla.org").*,
   FROM
     `moz-fx-data-marketing-prod.220432379.ga_sessions_*` AS ga_sessions
 ),
-ga_sessions_hits AS (
-  SELECT
```

Comment:
moved the fields formerly selected here into a UDF called before UNION ALL to make the query less sensitive to schema mismatch between the ga sessions for `vpn.mozilla.org` and `mozilla.org`